bam/runtime/src/main/java/io/quarkiverse/langchain4j/bam/runtime/BamRecorder.java

```diff
@@ -18,23 +21,44 @@ public Supplier<?> chatModel(Langchain4jBamConfig runtimeConfig) {
                 .timeout(runtimeConfig.timeout())
                 .logRequests(runtimeConfig.logRequests())
                 .logResponses(runtimeConfig.logResponses())
-
                 .modelId(chatModelConfig.modelId())
-                .version(chatModelConfig.version())
+                .version(runtimeConfig.version())
                 .decodingMethod(chatModelConfig.decodingMethod())
                 .minNewTokens(chatModelConfig.minNewTokens())
                 .maxNewTokens(chatModelConfig.maxNewTokens())
-                .temperature(chatModelConfig.temperature());
+                .temperature(chatModelConfig.temperature())
+                .includeStopSequence(firstOrDefault(null, chatModelConfig.includeStopSequence()))
+                .randomSeed(firstOrDefault(null, chatModelConfig.randomSeed()))
+                .stopSequences(firstOrDefault(null, chatModelConfig.stopSequences()))
+                .timeLimit(firstOrDefault(null, chatModelConfig.timeLimit()))
+                .topK(firstOrDefault(null, chatModelConfig.topK()))
+                .topP(firstOrDefault(null, chatModelConfig.topP()))
+                .typicalP(firstOrDefault(null, chatModelConfig.typicalP()))
+                .repetitionPenalty(firstOrDefault(null, chatModelConfig.repetitionPenalty()))
+                .truncateInputTokens(firstOrDefault(null, chatModelConfig.truncateInputTokens()))
+                .beamWidth(firstOrDefault(null, chatModelConfig.beamWidth()));
+
+        if (runtimeConfig.baseUrl().isPresent()) {
+            builder.url(runtimeConfig.baseUrl().get());
+        }
+
+        return () -> builder.build();
```

Comment:
Please avoid using lambdas in the `runtime` mave module.

Reply:
File updated without lambdas